### PR TITLE
Avoid data array overflow

### PR DESCRIPTION
**Pull request by seeers**
Occurs through packet collisions. Sometimes a wrong length is received.

### CODE REVIEW

**Comment by rejoe2** [comment]
Thanks for the patch!
Applied it yesterday to two of my nodes (V. 2.3.0-alfa). At first sight, this might solve the communication problems i ran into with my setup with 3 other Nodes+GW. 

As I also did some changes in the power supply of the nodes, possible root cause isn't limited to this patch.

**Comment by rejoe2** [comment]
In addition to my prior post:
Most likely this patch solves problems with colliding messages resulting in broken communication. After changing the power supply as mentionned, I kept 2 nodes with old code, they went offline after at about 18 hours (this was a rather long periode). After flashing the same hardware using this patch all (including other) nodes are online for more than 2.5 days now (some of them even 1.5 days longer).

Thanks a lot!

**Comment by seeers** [comment]
My pleasure. Now I just have to find out how to commit the patch :) I have not received any email with the patch as written by the Tollgate (Butler) ?

**Comment by rejoe2** [comment]
Sorry, but I'm also just a beginner in pull requests. The code itself seems ok imo, so no idea how to proceed. 
As some of the mods followed my thread in the forum (https://forum.mysensors.org/topic/7454/rs485-nodes-stop-sending-data-after-some-hours-or-day) and I referenced your patch there, it's perhaps easiest to ask for help there; most likely the mods can see more things in background or coding and help us (and others) to get that in the regular code base.

**Comment by fallberg** [comment]
@seeers please provide the patch sent by the butler (or add the missing space manually), amend the commit and force push (or create a new commit and push).

**Comment by fallberg** [comment]
The problem is that the coding standard mandates certain spacing. In this case the space after your function definition and the curly brace.

**Comment by fallberg** [comment]
@seeers you did not get a mail because you have not configured one in git. The Butler had no way of contacting you.

**Comment by fallberg** [comment]
Still whitespace errors. Most likely trailing spaces (can't see on phone). Link to patch in case you still have a missing email: https://ci.mysensors.org/job/MySensors/job/MySensors/job/PR-1142/4/execution/node/3/ws/MySensors/restyling.patch

File: hal/transport/RS485/MyTransportRS485.cpp [hunk 1]
```diff
@@ -154,6 +154,12 @@ bool _serialProcess()
 				}
 				_recPhase = 1;
 				_recPos = 0;
+
```
Comment (fallberg) [inline hal/transport/RS485/MyTransportRS485.cpp]: You have trailing whitespaces here.
That is what cause the butler to attempt to mail you a patch to remove them.

**Comment by fallberg** [comment]
I took the liberty of patching your PR. Please comment if you approve or reject the change. This should hopefully allow the tollgate to pass it.

**Comment by seeers** [comment]
It's ok, thank you!

**Comment by fallberg** [comment]
Thanks for the contribution!

**Pull request merged**